pkg/resolver/resolver.go

```diff
@@ -0,0 +1,61 @@
+package resolver
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/sonr-io/sonr/pkg/client"
+	"github.com/sonr-io/sonr/pkg/crypto/mpc"
+	"github.com/sonr-io/sonr/pkg/did"
+)
+
+type ResolverImpl struct {
+	wallet *mpc.Wallet
+	client *client.Client
+	did.Resolver
+}
+
+func New() did.Resolver {
+	return &ResolverImpl{}
+}
+
+// this could be abstracted into a provider for connections.
+// right now binded to the client pkg
+func (res *ResolverImpl) WithClient(client *client.Client) {
+	res.client = client
+}
+
+func (res *ResolverImpl) WithWallet(wallet *mpc.Wallet) {
+	res.wallet = wallet
+}
+
+func (r *ResolverImpl) Resolve(inputDID string) (*did.Document, *did.DocumentMetadata, error) {
+	if r.client == nil {
```

Comment:
If this struct needs a client to be useful, consider adding it to the `New` function to make it required.